site/app/libraries/Output.php

```diff
@@ -561,6 +561,24 @@ public function addInternalJs($file, $folder = 'js') {
         $this->addJs($this->timestampResource($file, $folder));
     }
 
+    /**
+     * @param array<mixed>$args
+     */
+    public function renderVue(string $page, array $args = []): string {
+        $this->addVendorJs(FileUtils::joinPaths('vue', 'vue.runtime.global.prod.js'));
+        $this->css->add($this->timestampResource('style.css', 'mjs/vue'));
+        $this->js->add($this->timestampResource('submitty-vue3-frontend.umd.js', 'mjs/vue'));
+
+        return "<script>" .
+                "window.addEventListener('load', () => {" .
+                    "window.submitty.render('[data-vuemount]', '$page', " . json_encode($args) . ");" .
+                "})" .
+            "</script>" .
+            "<div class=\"content\" data-vuemount>" .
+                "<noscript>You need Javascript enabled to see this page.</noscript>" .
+            "</div>";
+    }
```

Comment:
I think it would be better to render this via Twig.

Reply:
Implemented in https://github.com/Submitty/Submitty/pull/10715/commits/9a370b71dafbb0cf17c7779b9dd95d5a33ff9b0a